mono/mini/method-to-ir.c

```diff
@@ -3641,6 +3641,14 @@ handle_alloc (MonoCompile *cfg, MonoClass *klass, gboolean for_box, int context_
 	MonoInst *iargs [2];
 	void *alloc_ftn;
 
+	if (mono_class_get_flags (klass) & TYPE_ATTRIBUTE_ABSTRACT)	{
+		char* full_name = mono_type_get_full_name (klass);
+		cfg->exception_message = g_strdup_printf ("Cannot create an abstract class: %s", full_name);
+		g_free (full_name);
+		mono_cfg_set_exception (cfg, MONO_EXCEPTION_MEMBER_ACCESS);
```

Comment:
Could do
```c
		mono_cfg_set_exception (cfg, MONO_EXCEPTION_MONO_ERROR);
		mono_error_set_generic_error (&cfg->error, "System", "MethodAccessException", "Cannot crerate an abstract class: %s", full_name);
		g_free (full_name);
```

For bonus points add `mono_error_set_member_access (MonoError *error, const char *msg_format, ...) MONO_ATTR_FORMAT_PRINTF(2,3)` to `utils/mono-error-internals.h` analogously to `mono_error_set_invalid_program` and others.